pkg/utils/etcdutil/etcdutil.go

```diff
@@ -196,72 +210,237 @@ func EtcdKVPutWithTTL(ctx context.Context, c *clientv3.Client, key string, value
 	return kv.Put(ctx, key, value, clientv3.WithLease(grantResp.ID))
 }
 
-// CreateClients creates etcd v3 client and http client.
-func CreateClients(tlsConfig *tls.Config, acUrls url.URL) (*clientv3.Client, *http.Client, error) {
-	client, err := CreateEtcdClient(tlsConfig, acUrls)
-	if err != nil {
-		return nil, nil, errs.ErrNewEtcdClient.Wrap(err).GenWithStackByCause()
-	}
-	httpClient := createHTTPClient(tlsConfig)
-	return client, httpClient, nil
-}
+const (
+	// etcdServerOfflineTimeout is the timeout for an unhealthy etcd endpoint to be offline from healthy checker.
+	etcdServerOfflineTimeout = 30 * time.Minute
+	// etcdServerDisconnectedTimeout is the timeout for an unhealthy etcd endpoint to be disconnected from healthy checker.
+	etcdServerDisconnectedTimeout = 1 * time.Minute
+)
 
-// createEtcdClientWithMultiEndpoint creates etcd v3 client.
-// Note: it will be used by micro service server and support multi etcd endpoints.
-// FIXME: But it cannot switch etcd endpoints as soon as possible when one of endpoints is with io hang.
-func createEtcdClientWithMultiEndpoint(tlsConfig *tls.Config, acUrls []url.URL) (*clientv3.Client, error) {
-	if len(acUrls) == 0 {
-		return nil, errs.ErrNewEtcdClient.FastGenByArgs("no available etcd address")
-	}
-	endpoints := make([]string, 0, len(acUrls))
-	for _, u := range acUrls {
-		endpoints = append(endpoints, u.String())
-	}
+func newClient(tlsConfig *tls.Config, endpoints ...string) (*clientv3.Client, error) {
```

Comment:
No need to check the length of endpoints?

Reply:
I think only adding a check in CreateEtcdClient seems to be enough.

Reply:
only CreateEtcdClient will call it?

Reply:
another is `healthyChecker.addClient`, it will always pass an endpoint.

Reply:
Then what if we have some other functions call it later?